examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -300,6 +300,41 @@
 # Also, AGE, EXPERIENCE and EDUCATION are the three variables that most
 # influence the model.
 #
+# Interpreting coefficients: being cautious about causality
+# ---------------------------------------------------------
+#
+# Linear models are a great tool for measuring statistical association, but we
+# should be cautious when making statements about causality, after all
+# correlation doesn't always imply causation. This is particularly difficult in
+# the social sciences because the variables we observe only function as proxies
+# for the underlying causal process.
+#
+# In our particular case we can think of the EDUCATION of an individual as a
+# proxy for their professional aptitude, the real variable we're interested in
+# but can't observe. We'd certainly like to think that staying in school for
+# longer would increase technical competency, but it's also quite possible that
+# causality goes the other way too. That is, those who are technically
+# competent tend to stay in school for longer.
+#
+# An employer is unlikely to care which case it is (or if it's a mix of both),
+# as long as they remain convinced that a person with more EDUCATION is better
+# suited for the job, they will be happy to pay out a higher WAGE.
+#
+# This confounding of effects becomes problematic when thinking about some
+# form of intervention e.g. government subsidies of university degrees or
+# promotional material encouaring individuals to take up higher education.
+# The usefulness of these measures could end up being overstated, especially if
+# the degree of confounding is strong. Our model predicts a :math:`0.054699`
+# increase in hourly wage for each year of education. The actual causal effect
+# might be lower because of this confounding.
+#
+# Another problem when thinking about causality is that it may not make sense
+# to think about the causal effect of certain variables. In our example, while
+# it is perfectly valid to compare mean differences in WAGE based on race, it
+# does not make sense to think about the causal effect of race because (as far
+# as we know) it isn't possible to change one's race. Paul Holland coined the
+# phrase "no causation without manipulation" to reflect this idea.
```

Comment:
Thanks for the interesting addition. However, I think we might want to be more careful here. If I believe the point to still be valid, perhaps not quite with this example. At the intersection of causal inference and fairness, there is the notion of `counterfactual fairness` (see [this 2017 NeurIPS paper](https://papers.nips.cc/paper/2017/file/a486cd07e4ac3d270571622f4f316ec5-Paper.pdf) for instance) that specifically cover this type of situation.

Reply:
Hi, thanks for the feedback! 

My understanding is that counterfactual fairness is a valid concept when applied to race, but talking about a causal effect doesn't make sense because race as a variable cannot be manipulated. I haven't fully read Holland's [paper](https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.1086.7846&rep=rep1&type=pdf) but I'm reasonably sure that's his conclusion. 

At the very least we can change the wording to highlight the difficulties in interpreting a causal effect for a variable such as race without making such a strong conclusion. 

What do you think?

Reply:
If counterfactual fairness is not satisfied, that means there is a non-zero causal effect, so really the two concepts go together. Epistemologically, it's not because we cannot directly act or manipulate on a variable that we cannot have causal reasoning. And to some extent this is what causal inference is ultimately about. If you can directly act on a variable, then you are in the perfect RCT-world, do the experiment and measure the effect. Otherwise if action is impossible or limited, causal inference is a natural framework to see when and how we can proceed with causal effect estimation.

Reply:
I think I get what you're saying. Do you have any suggestions on how to rewrite that section or do you think I should remove it?

Reply:
IMHO it might a bit out of scope for the purpose of this documentation but I let the scikit-learn professionals chime in.

Reply:
Fair enough, happy to remove it but yeah, let's see what others might have to say.

Reply:
there are two issues here:

- I agree not being able to change a random variable, doesn't mean it can't have a causal effect;
- In particular talking about sensitive personal attributes such as race, we have social, ethical, and regulatory constraints to not have those causal relationships in our models, aka the model cannot be racist, sexist, etc.;
- Talking about causality and race also walks into the eugenics territory which we absolutely want to avoid.

So in the end. I think we're better off removing this paragraph, otherwise one needs to write an essay explaining why/when it's an issue working on causality of those variables with other constructs / measurements.

I'll remove this paragraph and merge, otherwise it's a very nice addition. Thank you.

cc @TamaraAtanasoska @glemaitre